driver/infoblox_ipam.go

```diff
@@ -76,10 +76,28 @@ func (ibDrv *InfobloxDriver) RequestAddress(r *ipamApi.RequestAddressRequest) (*
 	if rType, ok := r.Options["RequestAddressType"]; macAddr == "" || (ok && rType == netlabel.Gateway) {
 		logrus.Debugln("Request for Gateway IP")
 
+		l := &ibclient.NetworkViewLock{Name: network.NetviewName, ObjMgr: ibDrv.objMgr, LockEA: common.EA_DOCKER_PLUGIN_LOCK,
+			LockTimeoutEA: common.EA_DOCKER_PLUGIN_LOCK_TIME}
+		err := l.Lock()
+
+		if err != nil {
+			return  &ipamApi.RequestAddressResponse{}, fmt.Errorf("Failed to Allocate Gateway IP :%s", err)
+		}
+		defer l.UnLock(false)
+
+		if r.Address != "" {
+			ipAddr, err := ibDrv.objMgr.GetFixedAddress(network.NetviewName, network.Cidr, r.Address, "")
+			if err == nil && ipAddr != nil {
```

Comment:
> 1. So, what does docker think the MAC of the gateway is? Have you tested that network connection initiated by a docker container can be routed through the gateway?

Docker does not pass the MAC address. It just passes that IP address in the request. This is the request with gateway passed
```level=debug msg=\"RequestAddress Called : &{PoolID:network/ZG5zLm5ldHdvcmskMTcyLjI4LjEyOC4wLzI0LzI:172.28.128.0/24/local Address:172.28.128.1 Options:map[RequestAddressType:com.docker.network.gateway]}" plugin=0528e6bb30549584006f11ac5c01b4a8afa06246f5d88b0e82500fc208825376
```
I just tested that the container having the gateway configured as the gateway ip passed when creating the docker network.

> 2. And there's no way to inject the gateway MAC address to docker?

I dont think so there is a way to do that because the response for the RequestAddress call is just the IP address and IPAM specific metadata .
https://github.com/docker/libnetwork/blob/master/docs/ipam.md#requestaddress